iterative_verifier.go

```diff
@@ -573,7 +574,27 @@ func (v *IterativeVerifier) compareFingerprints(pks []uint64, table *schema.Tabl
 		return nil, targetErr
 	}
 
-	return compareHashes(sourceHashes, targetHashes), nil
+	mismatches := compareHashes(sourceHashes, targetHashes)
+	if len(mismatches) > 0 && v.CompressionVerifier != nil {
```

Comment:
I feel like this block of code here could be it's own function to make things more readable.
```go
mismatches := compareHashes(sourceHashes, targetHashes)

if mismatchInCompressedColumns(mismatches) {
  return compareCompressedColumns(...)
} else {
  return mismatches, nil
}
```
wdyt?

Reply:
That does look simpler, however, we would need to introduce some compression-specific logic to the `IterativeVerifier` to prevent duplication of code or flowing back and forth between the `CompressionVerifier` and `IterativeVerifier`. That would clean up this block of execution inside the conditional, but would blur the line of responsibility between these two components. We could do something like:
```go
if  v.compressionMismatch(mismatches) {
    sourceHashes, targetHashes, err = GetCompressedHashes(...)
    if err != nil {
        return nil, err
    }
    return compareHashes(sourceHashes, targetHashes), nil
}

func (v *IterativeVerifier) compressionMismatch(mismatches) {
    if len(mismatches) > 0 && v.CompressionVerifier != nil { 
        return true
    }
    return false
} 
```

and rename the current `GetCompressedHashes` to `GetCompressedTableHashes` and wrap the former to provide a clean interface.

Thoughts?